src/sentry/integrations/bitbucket_server/webhook.py

```diff
@@ -204,7 +210,12 @@ def post(self, request: HttpRequest, organization_id, integration_id) -> HttpRes
             interaction_type=event_handler.event_type,
             domain=IntegrationDomain.SOURCE_CODE_MANAGEMENT,
             provider_key=event_handler.provider,
-        ).capture():
-            event_handler(event, organization=organization, integration_id=integration_id)
+        ).capture() as lifecycle:
+            event_handler(
+                event,
+                organization=organization,
+                integration_id=integration_id,
+                lifecycle=lifecycle,
```

Comment:
could we move the declaration of the lifecycle to within the `event_handler`? Generally Idt we want to be combining observability and application logic in functions/bleeding that line

Reply:
makes sense to me 👍  good call

Reply:
updated, hiding whitespace will make this easier to review :+1: